src/plugin/PluginManager.php

```diff
@@ -626,8 +629,29 @@ public function registerEvents(Listener $listener, Plugin $plugin) : void{
 						throw new PluginException("Event handler " . Utils::getNiceClosureName($handlerClosure) . "() declares invalid @" . ListenerMethodTags::HANDLE_CANCELLED . " value \"" . $tags[ListenerMethodTags::HANDLE_CANCELLED] . "\"");
 				}
 			}
+			$noConcurrentCall = false;
+			if(isset($tags[ListenerMethodTags::NO_CONCURRENT_CALL])){
+				if(!is_a($eventClass, AsyncEvent::class, true)){
+					throw new PluginException(sprintf(
+						"Event handler %s() declares @%s for non-async event of type %s",
+						Utils::getNiceClosureName($handlerClosure),
+						ListenerMethodTags::NO_CONCURRENT_CALL,
+						$eventClass
+					));
+				}
+				switch(strtolower($tags[ListenerMethodTags::NO_CONCURRENT_CALL])){
+					case "true":
```

Comment:
since we have multiple of these, what about extracting to a function?

Reply:
This is not directly related to this PR, it should be done externally.